jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/config/JmxScraperConfig.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.jmxscraper.config;
+
+import java.util.Collections;
+import java.util.Set;
+
+/** This class keeps application settings */
+public class JmxScraperConfig {
+  String serviceUrl = "";
+  String customJmxScrapingConfigPath = "";
+  Set<String> targetSystems = Collections.emptySet();
+  int intervalMilliseconds;
+  String metricsExporterType = "";
+
+  String otlpExporterEndpoint = "";
+
+  String username = "";
+  String password = "";
+  String realm = "";
+  String remoteProfile = "";
+  boolean registrySsl;
```

Comment:
This feels like a symptom of groovy porting or something. I don't _really_ have an issue with it, it's just not idiomatic and will give java devs pause.

What do we think about making these private and adding getters, or maybe introducing `@AutoValue`?

Reply:
I have refactored and it's not using a static factory method instead of a separate class. I am not familiar with `@AutoValue` usage for such configuration classes so that could be another interesting option as well.